nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/LayerOpValidation.java

```diff
@@ -1492,50 +1528,65 @@ public void LSTMLayerTestCase1() {
+                SDVariable yLast = useYLast ? sd.var("yLast", Nd4j.zeros(DataType.DOUBLE, bS, numUnits)) : null;
 
-        LSTMLayerOutputs outputs = new LSTMLayerOutputs(sd.rnn.lstmLayer(
-                in, cLast, yLast, null,
-                LSTMLayerWeights.builder()
-                        .weights(sd.var("weights", Nd4j.rand(DataType.FLOAT, nIn, 4 * numUnits)))
-                        .rWeights(sd.var("rWeights", Nd4j.rand(DataType.FLOAT, numUnits, 4 * numUnits)))
-                        .peepholeWeights(sd.var("inputPeepholeWeights", Nd4j.rand(DataType.FLOAT, 3 * numUnits)))
-                        .bias(sd.var("bias", Nd4j.rand(DataType.FLOAT, 4 * numUnits))).build(),
-                c), c);
 
-        long[] out = new long[]{bS, numUnits, sL};
-        long[] hL = new long[]{bS, numUnits};
-        long[] cL = new long[]{bS, numUnits};
+                LSTMLayerConfig c = LSTMLayerConfig.builder()
+                        .lstmdataformat(LSTMDataFormat.NST)
+                        .directionMode(LSTMDirectionMode.FWD)
+                        .gateAct(LSTMActivations.SIGMOID)
+                        .cellAct(LSTMActivations.TANH)
+                        .outAct(LSTMActivations.TANH)
+                        .retFullSequence(true)
+                        .retLastC(true)
+                        .retLastH(true)
+                        .build();
 
-        assertArrayEquals(out, outputs.getOutput().eval().shape());
-        assertArrayEquals(hL, outputs.getLastTimeStepOutput().eval().shape());
-        assertArrayEquals(cL, outputs.getLastCellStateOutput().eval().shape());
+                LSTMLayerOutputs outputs = new LSTMLayerOutputs(sd.rnn.lstmLayer(
+                        in, cLast, yLast, null,
+                        LSTMLayerWeights.builder()
+                                .weights(sd.var("weights", Nd4j.randn(DataType.DOUBLE, nIn, 4 * numUnits)))
+                                .rWeights(sd.var("rWeights", Nd4j.randn(DataType.DOUBLE, numUnits, 4 * numUnits)))
+                                .peepholeWeights(sd.var("inputPeepholeWeights", Nd4j.randn(DataType.DOUBLE, 3 * numUnits)))
+                                .bias(sd.var("bias", Nd4j.rand(DataType.DOUBLE, 4 * numUnits))).build(),
+                        c), c);
+
+                long[] out = new long[]{bS, numUnits, sL};
+                long[] hL = new long[]{bS, numUnits};
+                long[] cL = new long[]{bS, numUnits};
+
+                assertArrayEquals(out, outputs.getOutput().eval().shape());
+                assertArrayEquals(hL, outputs.getLastOutput().eval().shape());
+                assertArrayEquals(cL, outputs.getLastState().eval().shape());
+
+                sd.setLossVariables(outputs.getOutput(), outputs.getLastTimeStepOutput(), outputs.getTimeSeriesOutput());
+
+                String err = OpValidation.validate(new TestCase(sd)
+                        .gradientCheck(true)
```

Comment:
One small hting we can do to help us debug (if it fails on CI) so we know which test case:
add `.testName("cLast=" + cLast + ", yLast=" + yLast)` to TestCase